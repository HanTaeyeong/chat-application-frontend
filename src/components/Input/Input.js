import React from 'react';

import './Input.scss'

export default function Input({ text, setText, sendMessage }) {
    const onClickButton = (e) => {
        e.preventDefault()
        sendMessage()
    }
    return <form className='form'>
        <input
            id='message-input'
            type='text'
            placeholder='Type a message'
            value={text}
            onChange={e => setText(e.target.value)}
            onKeyPress={e => e.key === 'Enter' && sendMessage()}
        />
        <button className='send-button' onClick={e => onClickButton(e)}>Send</button>

    </form>;
}
