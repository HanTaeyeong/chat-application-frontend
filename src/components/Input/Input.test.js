import React from 'react'
import Input from './Input'

import {create} from 'react-test-renderer'


describe('Input Component Test', () => {
    test('render test', () => {   
        let app=create(<Input />) 
        expect(app.toJSON()).toMatchSnapshot()
    })
})