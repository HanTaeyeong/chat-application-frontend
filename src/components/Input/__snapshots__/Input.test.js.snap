// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Test render test 1`] = `
<form
  className="form"
>
  <input
    id="message-input"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Type a message"
    type="text"
  />
  <button
    className="send-button"
    onClick={[Function]}
  >
    Send
  </button>
</form>
`;
